src/util/each.js

```diff
@@ -18,7 +18,7 @@ export function eachNodeOrFragmentNodes(node, func) {
     const chs = node.childNodes;
     const chsLen = chs.length;
     for (let a = 0; a < chsLen; a++) {
-      func(chs[a], a);
+      func(node.cloneNode(true).childNodes[a], a);
```

Comment:
Yep, every shadow root's content instead of every node :) not sure it's much better